lib/net/sasl/anonymous_authenticator.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Net
+
+  module SASL
+
+    # Authenticator for the "+ANONYMOUS+" SASL mechanism, specified in
+    # RFC4616[https://tools.ietf.org/html/rfc4505].
+    class AnonymousAuthenticator < Authenticator
+
+      # This should generally be instantiated via Net::SASL.authenticator.
+      def initialize(*)
+        super
+      end
+
+      # There are no responses for ANONYMOUS
+      def process(_data)
+          raise ArgumentError, "SASL ANONYMOUS does not produce any responses"
```

Comment:
This isn't quite right.  A successful SASL exchange always has at least one response.  If the protocol, the server's capabilities, and the mechanism all support an "initial response", `process` will be called with `nil`. If not, it'll be called with the server's "challenge" (which should always be empty for `ANONYMOUS`).

`ANONYMOUS` should respond with a string of zero to 255 UTF-8-encoded Unicode characters.  That could simply be a `@message` ivar set in the initializer.  Or even the `@username` ivar that's already set by the superclass.

Reply:
Ah, right, I implemented this based on its usage rather than reading the RFC, bad me.  I've changed this now.

Reply:
Thanks! I have another small change or two I'm considering, based on _(unreleased)_ changes to the superclass.  But there's no guarantee I'll get to those soon! 😉